### PR TITLE
API Deprecate CwpSiteSummaryExtension

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/cwp/issues/320

### CODE REVIEW

File: src/Extensions/CwpSiteSummaryExtension.php [hunk 1]
```diff
@@ -3,12 +3,24 @@
...
 /**
  * Extends the site summary report to list the appropriate versions in the report header
+ *
+ * @deprecated 4.13.0 Will be removed without equivalent functionality to replace it
  */
 class CwpSiteSummaryExtension extends Extension
 {
+    public function __construct()
+    {
+        Deprecation::notice(
```
Comment (ChloeSartorelli) [inline src/Extensions/CwpSiteSummaryExtension.php]: Should be wrapped in `withNoReplacement()` since we're still using this in CMS 4.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Extensions/CwpSiteSummaryExtension.php (hunk 1 above)
Comment (emteknetnz) [reply src/Extensions/CwpSiteSummaryExtension.php]: Updated

Comment (edwilde) [reply src/Extensions/CwpSiteSummaryExtension.php]: Heads up, this just created an error for me when using with `silverstripe/framework@4.12.3`
> Error: Call to undefined method SilverStripe\\Dev\\Deprecation::withNoReplacement()

([ref](https://github.com/silverstripe/silverstripe-framework/blob/4.12.3/src/Dev/Deprecation.php))

Comment (ChloeSartorelli) [reply src/Extensions/CwpSiteSummaryExtension.php]: @edwilde Thanks, I'll update the dependency for `2.11` of this module.

**Pull request merged**